src/Database/LSMTree/Internal/Run/Mutable.hs

```diff
@@ -51,29 +54,34 @@ type RefCount = IORef Int
 -- representation of a run. Each file handle is opened write-only and should be
 -- written to using normal buffered I\/O.
 --
+-- __Not suitable for concurrent construction from multiple threads!__
+--
```

Comment:
Makes sense -- we can wrap the `MRun` type in something to handle concurrent access